### ISSUE TITLE
Look into how to enforce two-factor authentication on AWS while still being able to use the AWS CLI

### ISSUE THREAD

**Issue by forabi** [hollowverse-archive/hollowverse#249]

**Comment by forabi** [comment]
So here's what we have currently:

* An IAM policy that requires MFA logins for all AWS actions, except actions that allow users to configure their MFA settings.
* An IAM group called "MFA-Enforced" which should be manually assigned to any human users. This group enforces the above policy.
* The reason why it should not be assigned to bot users is that these accounts won't be able to do anything because they won't be signed in with MFA.
* Adding human users to "MFA-Enforced" prevents regular CLI access because this access is similar to a login without a MFA code.
* However, I was able to use the CLI even when the MFA policy is enforced on my IAM user by following [the instructions here](https://aws.amazon.com/premiumsupport/knowledge-center/authenticate-mfa-cli/). It's a little bit inconvenient because I need to execute the command to get a "session token" every 12-36 hours, and manually export the new secrets as variables in my shell session.
* This inconvenience can be avoided with a tool like [`aws-mfa`](https://github.com/broamski/aws-mfa) which automates this process.


So to sum things up:

* For any new user of our AWS account, we should manually assign them to "MFA-Enforced".
* Non-humans users should not be assigned to that group.
* For users that need access to the AWS CLI, recommend `aws-mfa` to automate the process of updating the authentication details every 12-36 hours.

**Comment by wholesomedev** [comment]
@forabi Cool. Thanks for doing this research. It clarifies things. 

So what's the next action item on this issue?

**Comment by forabi** [comment]
I think we can close it now :tada:
